### DISCUSSION TITLE
Help wanted: Setting up on Hostgator

### DISCUSSION THREAD

**Discussion by bc3tech** [shlinkio/shlink#1972] [Help wanted] [answered]
### Shlink version

3.7.3

### PHP version

8.3

### How do you serve Shlink

Self-hosted Apache

### Database engine

MySQL

### Database version

5.7.23-23

### Summary

Hey folks,

I'm trying to get shlink set up on my HostGator shared hosting account.
I've d/l'd and unzipped the package into a new site directory, configured via cpanel.
Then I ran the installer script via the terminal, configuring access to mySql - everything succeeds as expected; the database is set up with all the tables, etc. I elected not to wire up GeoLite2:

![image](https://github.com/shlinkio/shlink/assets/1967433/fc63e3f1-b82f-4668-9314-a14a3e9f5d54)

![image](https://github.com/shlinkio/shlink/assets/1967433/f138f92f-992c-459a-badb-abd327768eb3)

I can run `bin/cli` successfully, and have API keys generated.

![image](https://github.com/shlinkio/shlink/assets/1967433/65e1e1a3-1312-4b75-bb38-fc1e4d49b299)

The piece I think I'm missing is perhaps something to go into the `.htaccess` file at the root of the site.

![image](https://github.com/shlinkio/shlink/assets/1967433/c5d4ab0e-1b69-41f1-9c6e-d6cfa6d05eb5)

The problem I'm seeing is every request, no matter what subdir/path, from a browser or PostMan, all return 404. Of course, the shlink.io site says it can't use my installation either. I tried adding [the `VirtualApplication` piece from the docs](https://shlink.io/documentation/supported-runtimes/classic-web-server/#apache) but saw no change.

Can somebody help me out in determining what I need to do to get this last piece wired up?

Thanks!

**Comment by acelaya** [comment, marked as answer]
How do those 404 "look like"?

Depending on that, it could be apache not pointing to the right place, or something else.

Also, make sure the document root is configured to point to the `public` directory, not Shlink's root folder.

**Reply by bc3tech** [reply]
rock on. I'll double-check all this. if it still doesn't work i'll post a screenshot of the 404. I just nuked it all to start over and try again :)

**Reply by bc3tech** [reply]
Got it working! I think the kicker was having docroot -> `/public` - thanks!

Question: Is the _only_ admin web interface for an shlink install via app.shlink.io? No admin portal present on the installation itself?

Thanks again!

**Reply by acelaya** [reply]
> Question: Is the only admin web interface for an shlink install via app.shlink.io? No admin portal present on the installation itself?

Correct. It's explained here https://shlink.io/documentation/advanced/shlink-architecture/

What you can do is host shlink-web-client yourself, if you want, but if you don't have any particular requirements or restrictions, I would recommend just sticking with https://app.shlink.io, and that's one less thing to worry about.
